### PR TITLE
Feature/bonsai asset

### PR DESCRIPTION
**Pull request by jspaleta**
#### General

- [X] Update Changelog following the conventions laid out [here](https://github.com/sensu-plugins/community/blob/master/HOW_WE_CHANGELOG.md)

- [X] Update README with any necessary configuration snippets

- [ ] Binstubs are created if needed

- [X] RuboCop passes

- [X] Existing tests pass

#### Purpose
Build Sensu assets using sensu-ruby-runtime as part of travisCI release tagging.

#### Additional Changes
update gemspec to use latest sensu-plugin 
update gemspec and travis to drop old ruby no longer supported in travisCI.

### CODE REVIEW

**Comment by jspaleta** [comment]
Note 1:
I'm expecting the before_deploy   command to change again slightly, I have a pending PR against the helper repository. Once that is merged I can change the before_deploy to pull master branch, or specific commit instead of the feature branch.

Note 2:
I'm using travis encrypt to save a personal oauth GITHUB_TOKEN used to upload binaries into the github release as part of the asset deploy script,   This token might need to change if there is a sensu-plugins org-wide preferred token to use here.

**Comment by jspaleta** [comment]
Okay I think this is good to go.

Note the new bonsai badge committed as part of the README won't work until after this PR is merged and a new release is tagged.  I'll need to manually add this plugin into Bonsai after this PR is merged and a new release tagged.

**Comment by jspaleta** [comment]
For an example of what the hew travis config does, you can look at my fork. I've built a 4.0.2-pre tag as part of my repository fork firing off the new asset build and release logic.

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,9 +4,18 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 This CHANGELOG follows the format listed [here](https://github.com/sensu-plugins/community/blob/master/HOW_WE_CHANGELOG.md)
 
 ## [Unreleased]
+### Breaking Changes
+- Update minimum required ruby version to 2.3. Drop unsupported ruby versions.
+- Update dependancy on sensu-plugin to 4.0
```
Comment (majormoses) [inline CHANGELOG.md]: lets make sure we have links to the relevant changelog so people can more easily evaluate their upgrade path. Here is an example:
```
- bump `sensu-plugin` dependency from `~> 1.2` to `~> 4.0` you can read the changelog entries for [4.0](https://github.com/sensu-plugins/sensu-plugin/blob/master/CHANGELOG.md#400---2018-02-17), [3.0](https://github.com/sensu-plugins/sensu-plugin/blob/master/CHANGELOG.md#300---2018-12-04), and [2.0](https://github.com/sensu-plugins/sensu-plugin/blob/master/CHANGELOG.md#v200---2017-03-29)
```

File: .travis.yml [hunk 2]
```diff
@@ -16,14 +18,25 @@ script:
...
+- provider: script
+  script: bonsai/ruby-runtime/travis-build-ruby-plugin-assets.sh sensu-plugins-disk-checks
+  skip_cleanup: true
+  on:
+    tags: true
+    all_branches: true
+    rvm: 2.4.1
+env:
+  global:
+    secure: mayZ2ZMGx0v/efTaupGXs79ni272J2mjoSC6wq+YPbeTw/BV1G/oPXja83Wxnyb1XOzuWHwH5xQUv91luF8KPJp56Xafwczi96LdZx+osYavcpGo+x1u7e4YDG0TSDB5f1elMAraiMHsQPR2TQqCYXQRj16zznCX5nPVAlq+FZPYWPeoNtBLajA68ou4vNVSKswE75OFEvU1DzSHtxQ/Q3biXseZ0puhb+oyV6798l4N0tbe3NHwPaVzsFjXgacg8QCqgtooR2nVAPx3918tyNJeKJHi7RN1h0Fm25t1O5hxwqv9iPtJNh4uF9P1RASnW4a3e0Vc41HenK51AvsjutTeoSudAyCMt0YInT91a1c+9vxXppmhsknCBDy0WUMttE1MLXUgOyG9+xsTuEjnP0JoYG8Y16vEIlIvwx+vo4IqP/YIesbNZ+x14v8pWF8VxklRV7DU74llh9lecffXoG/rHj9l2evOLch1JX8A59p/JdJ4RmIitlvRg90OULa2M6Q6R52In/+CLso8z/ZbRblo6Dj33daJdLupazgTqLGNaCYxfAUOfGblp2piFpiEgYsf1JWptFwwECaIyka0saW3H0ygMUjMt6uFjkcDocchbiY/dV1T9GsX0w79Z2LJkosnpVmpheN11ICMkwZR2Qdw/HBoKbX5PHJUn1rG3Cs=
```
Comment (majormoses) [inline .travis.yml]: this encrypted value is for a token connected to whom? We should probably generate one with the `sensu-plugin` user with a limited oauth scope, which scopes did you give your token?

File: .bonsai.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
+---
```
Comment (majormoses) [inline .bonsai.yml]: I'm wondering if we could maybe have something similar to rubocop that allows us to pull in common attributes from a single location so its quick to update.

File: .travis.yml (hunk 2 above)
Comment (jspaleta) [reply .travis.yml]: yeah right now this is associated with my account.  

Public_repo scope is all this OATH personal token uses.

File: .bonsai.yml (hunk 3 above)
Comment (jspaleta) [reply .bonsai.yml]: hmm.  Interesting.  We could have the sensu-go-bonsai-asset repo as a submodule or subtree.

File: CHANGELOG.md (hunk 1 above)
Comment (jspaleta) [reply CHANGELOG.md]: I'll update the changelog accordingly

File: .travis.yml (hunk 2 above)
Comment (jspaleta) [reply .travis.yml]: we may need to create a "machine account" that is a member of the sensu-plugins org that we can then use for automation.  Github only lets you generate oath tokens for "accounts" not "orgs"  but it looks like the best practice pattern here is to create a dedicated github account not tied to a specific person, that can be used to do org  automation tasks.

That way if a person drops out of the org, you don't have to retool the tokens associated with their account.

File: CHANGELOG.md (hunk 1 above)
Comment (jspaleta) [reply CHANGELOG.md]: updated in latest commit to this PR branch

File: .travis.yml (hunk 2 above)
Comment (majormoses) [reply .travis.yml]: yes this is what the `sensu-plugin` (no `s`) user is.

Comment (jspaleta) [reply .travis.yml]: Okay I set up the repo in travisCI's webUI to use sensu-plugin token as part of repository settings..similar to how codeclimate token is configured.

File: .bonsai.yml (hunk 3 above)
Comment (jspaleta) [reply .bonsai.yml]: I'll look at this as a future enhancement.  One problem here is that bonsai currently needs the .bonsai.yml to exist in the master branch to register the repository. So I'm not sure we can automate this completely.

**Comment by sensu-discourse** [comment]
This pull request has been mentioned on **Sensu Community**. There might be relevant details there:

https://discourse.sensu.io/t/how-to-use-the-sensu-plugins-kubernetes-plugin/1286/2

**Pull request merged**